frontend/src/metabase/visualizations/echarts/pie/option/index.ts

```diff
@@ -0,0 +1,222 @@
+import Color from "color";
+import type d3 from "d3";
+import type { EChartsOption } from "echarts";
+import cloneDeep from "lodash.clonedeep";
+
+import { getTextColorForBackground } from "metabase/lib/colors";
+import type {
+  ComputedVisualizationSettings,
+  RenderingContext,
+} from "metabase/visualizations/types";
+
+import { DIMENSIONS, TOTAL_TEXT } from "../constants";
+import type { PieChartFormatters } from "../format";
+import type { PieChartModel, PieSlice, PieSliceData } from "../model/types";
+
+import { SUNBURST_SERIES_OPTION, TOTAL_GRAPHIC_OPTION } from "./constants";
+
+function getSliceByKey(key: PieSliceData["key"], slices: PieSlice[]) {
+  const slice = slices.find(s => s.data.key === key);
+  if (!slice) {
+    throw Error(
+      `Could not find slice with key ${key} in slices: ${JSON.stringify(
+        slices,
+      )}`,
+    );
+  }
+
+  return slice;
+}
+
+function getTotalGraphicOption(
+  chartModel: PieChartModel,
+  formatters: PieChartFormatters,
+  renderingContext: RenderingContext,
+  hoveredIndex: number | undefined,
+  outerRadius: number,
+) {
+  const graphicOption = cloneDeep(TOTAL_GRAPHIC_OPTION);
+
+  // Don't display any text if there isn't enough width
+  if (outerRadius * 2 < DIMENSIONS.totalDiameterThreshold) {
+    return graphicOption;
+  }
+
+  graphicOption.children.forEach(child => {
+    child.style.fontFamily = renderingContext.fontFamily;
+  });
+
+  graphicOption.children[0].style.text = formatters.formatMetric(
```

Comment:
I'd prefer having `getTotalGraphicOption` to define and return an object instead of cloning `TOTAL_GRAPHIC_OPTION` and then mutating its properties because this approach in my opinion is less readable because now a relatively small object construction is spread in two places and to understand what elements `graphicOption.children[0]` and `graphicOption.children[1]` are I need to go to the definition of `TOTAL_GRAPHIC_OPTION`. Same for `SUNBURST_SERIES_OPTION`

Reply:
Sure I can change that

Reply:
Done